jupyterhub_moss/form/option_form.js

```diff
@@ -1,7 +1,215 @@
+  const hidden_input = document.createElement('input');
+  hidden_input.setAttribute('type', 'hidden');
+  hidden_input.setAttribute('value', key);
+  div.appendChild(hidden_input);
+
+  const radio_id = `environment_radio_${key}`;
+  const title = `Environment path: ${path}`
+
+  const input = document.createElement('input');
+  input.setAttribute('type', 'radio');
+  input.setAttribute('id', radio_id);
+  input.setAttribute('title', title)
+  input.setAttribute('name', 'environment_path');
+  input.setAttribute('value', path);
+  if (checked) {
+    input.setAttribute('checked', '');
+  }
+  div.appendChild(input);
+
+  const label = document.createElement('label');
+  label.setAttribute('for', radio_id);
+  label.setAttribute('title', title);
+  label.textContent = description === '' ? path : description;
+  div.appendChild(label);
+
+  return div;
+}
+
+function resetEnvironmentSelection() {
+  const environmentsDiv = document.getElementById('jupyter_environments');
+  const environmentSimpleSelect = document.getElementById('environment_simple')
+
+  environmentsDiv.querySelector('input[type="radio"]').checked = true;
+  environmentSimpleSelect.selectedIndex = 0;
+}
+
+function selectEnvironment(key) {
+  const environmentsDiv = document.getElementById('jupyter_environments');
+  const environmentSimpleSelect = document.getElementById('environment_simple');
+
+  if (key !== null) {
+    const keyInputs = environmentsDiv.querySelectorAll('input[type="hidden"]');
+    for (let index = 0; index < keyInputs.length; index++) {
+      if (keyInputs[index].value === key) {
+        keyInputs[index].parentNode.querySelector('input[type="radio"]').checked = true;
+        environmentSimpleSelect.selectedIndex = index;
+        return;
+      }
+    }
+  }
+  resetEnvironmentSelection();
+}
+
+function getSelectedEnvironment() {
+  const environmentsDiv = document.getElementById('jupyter_environments');
+
+  // Get selected environment if any
+  const selectedRadio = environmentsDiv.querySelector('input[type="radio"]:checked');
+  if (selectedRadio !== null) {
+    const hiddenInput = selectedRadio.parentNode.querySelector('input[type="hidden"]');
+    if (hiddenInput !== null) {
+      return hiddenInput.value;
+    }
+  }
+  return null;
+}
+
+function addCustomEnvironment(key, description, path, persist=true) {
+  const customEnvironmentDiv = document.getElementById('jupyter_environments_custom');
+  const environmentSimpleCustomOptGroup = document.getElementById('environment_simple_custom');
+
+  const div = createEnvironmentDiv(key, description, path);
+
+  button = document.createElement('button');
+  button.setAttribute('type', 'button');
+  button.setAttribute('title', 'Remove this environment from the custom environments');
+  button.setAttribute('value', key);
+  button.innerHTML = '&#xff0d;';
+  button.addEventListener(
+    'click', e => removeCustomEnvironment(e.target.value)
+  )
+  div.appendChild(button);
```

Comment:
I would add some CSS to the remove button so that it is a bit less prominent compared to the adding button

Reply:
I reduced height of the remove buttons.

<img width="471" alt="Screen Shot 2022-01-14 at 10 46 14" src="https://user-images.githubusercontent.com/9449698/149494917-710fbd01-5597-4bde-98be-0a14c43c6a43.png">